types/twilio-client/index.d.ts

```diff
@@ -0,0 +1,1065 @@
+         * This is triggered when an incoming connection is canceled by the caller before it is accepted by the Twilio Client device.
+         */
+        on(event: cancelEvent, handler: (connection: Connection) => any): this;
+
+        /**
+         * This is triggered when a connection is opened, whether initiated using `.connect()` or via an
+         * [accepted incoming connection](https://www.twilio.com/docs/voice/client/javascript/device#incoming).
+         *
+         * Since this Connection object represents an inactive connection, you'll probably just want to use the
+         * event to update your application's UI, then throw away the Connection and wait for the next call.
+         */
+        on(event: connectEvent, handler: (connection: Connection) => any): this;
+
+        /**
+         * Fired any time a {@link Connection} is closed. The handler function receives the `Twilio.Connection` object
+         * that was just closed as an argument.
+         */
+        on(
+            event: disconnectEvent,
+            handler: (connection: Connection) => any
+        ): this;
+
+        /**
+         * Emitted when any device error occurs. These may be errors in your request, your token, connection errors,
+         * or other application errors. Using the error handler is a great way to debug your application and help catch
+         * mistakes in your code.
+         */
+        on(event: errorEvent, handler: (error: DeviceError) => any): this;
+
+        /**
+         * This is triggered whenever an incoming connection from an [outbound REST call](https://www.twilio.com/docs/api/rest/making_calls#example-2) or a
+         * TwiML `<Dial>` to this device is made.
+         *
+         * This connection will be in state `pending` until you call `.accept()` on it.
+         */
+        on(
+            event: incomingEvent,
+            handler: (connection: Connection) => any
+        ): this;
+
+        /**
+         * This is triggered when the connection to Twilio drops or the device's token is invalid/expired.
+         * In either of these scenarios, the device cannot receive incoming connections or make outgoing connections.
+         * If the token expires during an active connection the `offline` event will be fired, but the connection will not be
+         * terminated. In this situation you will have to call `Twilio.Device.setup()` with a valid token before attempting
+         * or receiving the next connection.
+         */
+        on(event: offlineEvent, handler: (device: Device) => any): this;
+
+        /**
+         * This is initially triggered when all operations in `.setup()` have completed and the device is
+         * ready and online. It may be triggered again if the device goes offline and comes back online (i.e. theconnection drops and returns).
+         */
+        on(event: readyEvent, handler: (device: Device) => any): this;
+
+        toString(): string;
+    }
+}
+
+export default Twilio;
```

Comment:
I don't think this is right. Looking [at the source at es5/twilio.js](https://unpkg.com/twilio-client@1.6.5/es5/twilio.js), there is no default export, just Device, PStream, Connection, etc.

Reply:
I looked at the docs and it actually looks like the *usual* usage is in the browser, without modules at all. You just have a namespace Twilio. Here's the syntax to enable both commonjs imports and global browser usage:

```ts
export = Twilio; // commonjs
export as namespace Twilio; // global browser
```

Then you have to import Twilio in Typescript with `import Twilio = require('twilio-client')`. THat's equivalent to the JS `const Twilio = require('twilio-client')`. I'm pretty sure that's what Babel (for example) will emit for `import Twilio from 'twilio-client'`. 

Typescript will do that too, but you have to set esModuleInterop: true, which is not guaranteed to be on. So for compatibility, Definitely Typed packages need to use `export =` instead of `export default`

Reply:
Ok cool, ill make those updates later today, thanks for the help @sandersn !